test/test_02_04.py

```diff
@@ -0,0 +1,43 @@
+from unittest.mock import Mock
+
+from hm_1.tasks.task_02_04 import cache
+
+
+def func(a, b):
+    return (a ** b) ** 2
+
+
+mock = Mock(side_effect=func)
```

Comment:
Move this line inside each test. And call mock.assert_called_once to ensure a single call.

Mock is mutated during calls, you do not want all tests to depend on a single mutable object, each test should have its own mock.